src/pybind/mgr/dashboard/tools.py

```diff
@@ -910,3 +915,31 @@ def find_object_in_list(key, value, iterable):
         if key in obj and obj[key] == value:
             return obj
     return None
+
+
+def update_dict(data, update_data):
```

Comment:
Can you please add more tests like
```
>>> update_dict({'foo': {'bar': 5}}, {'foo.xyz': 10})
{'foo': {'bar': 5, 'xyz': 10}}
```

Reply:
> Calling the function with `update_dict({'foo': {'bar': 5}}, {'xyz': 10})` throws an error. I expect the result of `{'foo': {'bar': 5}, 'xyz': 10}`.

Why do you expect that? It doesn't state that it does that. Of course it could do that, but that's not the use case I had in mind when writing the tests. I **want** to have an exception thrown in a test when I accidentally add a key to a dict instead of updating it's value.

But if the function should be used differently at some point, because you want to use it in the backend code, or the tests need that requested different behavior, which I would love to see, then I think it's the time to extend it and ensure, that it works for tests as it currently does, because that's intended.

Reply:
> Calling the function with `update_dict({'foo': {'bar': 5}}, {'xyz': 10})` throws an error. I expect the result of `{'foo': {'bar': 5}, 'xyz': 10}`.

Why do you expect that? It doesn't state that it does that. Of course it could do that, but that's not the use case I had in mind when writing the tests. I **want** to have an exception thrown in a test when I accidentally add a key to a dict instead of updating it's value.

But if the function should be used differently at some point, because you want to use it in the backend code, or the tests need that requested different behavior, which I would love to see, then I think it's the time to extend it and ensure, that it works for tests as it currently does, because that's intended.

Reply:
> Can you please add more tests like
> 
> ```
> >>> update_dict({'foo': {'bar': 5}}, {'foo.xyz': 10})
> {'foo': {'bar': 5, 'xyz': 10}}
> ```

I could add the test, but the result wouldn't become what you expect.

Reply:
The function name and doc tells me that i can update a dictionary. If you WANT to get an exception when you insert values that will fail to update the origin dict, then please add a comment that this is by purpose. Maybe use ``:raises`` for that.